core/src/main/java/org/frankframework/http/rest/ApiServiceDispatcher.java

```diff
@@ -74,54 +75,196 @@ public static synchronized ApiServiceDispatcher getInstance() {
+		return findMatchingConfigsForUri(uri, false);
+	}
+
+	/**
+	 * Find an {@link ApiDispatchConfig} that has an exact match with the request URI, for purpose
+	 * of generating the OpenAPI spec from it.
+	 * Therefor no dispatch configurations for matching-patterns ending with {@code /**} are returned.
+	 *
+	 * @param uri The full URI for which to generate an OpenAPI spec.
+	 * @return The {@link ApiDispatchConfig} from which to generate an OpenAPI spec.
+	 */
+	@Nullable
+	public ApiDispatchConfig findExactMatchingConfigForUri(@Nonnull String uri) {
 		List<ApiDispatchConfig> configs = findMatchingConfigsForUri(uri, true);
-		return configs.isEmpty()? null : configs.get(0);
+		return configs.stream()
+				.filter(cfg -> !cfg.getUriPattern().endsWith("/**"))
+				.max(Comparator.comparingInt(cfg -> scoreUriPattern(cfg.getUriPattern())))
+				.orElse(null);
 	}
 
-	public List<ApiDispatchConfig> findMatchingConfigsForUri(String uri) {
-		return findMatchingConfigsForUri(uri, false);
+	/**
+	 * Find the {@link ApiDispatchConfig} best matching a given request, consisting of the
+	 * HTTP request method and request URI.
+	 * <p>
+	 *     This method will return the {@link ApiDispatchConfig} that has the most specific match
+	 *     with the request URI, and supports the requested HTTP method.
+	 * </p>
+	 * <p>
+	 *     {@link org.frankframework.http.rest.ApiListener.HttpMethod#OPTIONS} requests and requests for methods that are not supported by the configuration are always matched to the most specific URI pattern match.
+	 *     This is so that the {@code OPTIONS} request will return correct results, and requests for unsupported methods will return HTTP status code
+	 *     {@code 405} instead of {@code 404}.
+	 * </p>
+	 * <p>
+	 *     So for instance if a configuration would have the following {@link ApiListener}s installed:
+	 *     <lu>
+	 *         <li>ApiListener1: GET on uri /user/**</li>
+	 *         <li>ApiListener2: GET on uri /user/{userId}/department/{departmentId}</li>
+	 *         <li>ApiListener3: POST on uri /user/{userId}/avatar</li>
+	 *     </lu>
+	 *     Then:
+	 *     <lu>
+	 *         <li>A request {@code GET /user/usr123/department/dept456} would return the {@link ApiDispatchConfig} for /user/{userId}/department/{departmentId} containing ApiListener2</li>
+	 *         <li>A request {@code OPTIONS /user/usr123/department/dept456} would return the {@link ApiDispatchConfig} for /user/{userId}/department/{departmentId} containing ApiListener2</li>
+	 *         <li>A request {@code GET /user/usr123/avatar} would return the {@link ApiDispatchConfig} for /user/** containing ApiListener1</li>
+	 *         <li>A request {@code POST /user/usr123/avatar} would return the {@link ApiDispatchConfig} for /user/{userId}/avatar containing ApiListener3</li>
+	 *         <li>A request {@code PUT /user/usr123/avatar} would return the {@link ApiDispatchConfig} for /user/{userId}/avatar containing ApiListener3</li>
```

Comment:
But put has not been configured?

Reply:
And that is the point. I'll add more explicit explanation that this is for error-reporting purposes.